io/plugins/eu.esdihumboldt.hale.io.csv/src/eu/esdihumboldt/hale/io/csv/writer/AbstractTableInstanceWriter.java

```diff
@@ -146,6 +146,45 @@ else if (useSchema) {
 
 	} // close method
 
+	/**
+	 * Iterates over properties of the type definition and creates a map of the
+	 * given properties
+	 * 
+	 * @param definition given type to check
+	 * @param headerRow the current header row of the table
+	 * @param solveNestedProperties <code>true</code> if nested properties
+	 *            should be solved, otherwise <code>false</code>
+	 * @param useSchema <code>true</code> if properties should be defined from
+	 *            the schema, otherwise <code>false</code> and properties are
+	 *            defined from the Instances, but for the type cannot be taken
+	 *            into consideration as there is no instance available
+	 * @return a map of properties with string of localpart of the QName of the
+	 *         property as key
+	 */
+	protected Map<String, Object> getPropertyMap(TypeDefinition definition, List<String> headerRow,
+			boolean useSchema, boolean solveNestedProperties) {
```

Comment:
`useSchema` - this is not really a variable here, because you work based on the schema in any case.
`solveNestedProperties` - similar here, since you only look at top level properties and not what kind of nested properties there might be. Which is perfectly fine for the use case.